internal/server/proto/server.proto

```diff
@@ -102,6 +102,9 @@ service Waypoint {
   // each variable.
   rpc GetConfig(ConfigGetRequest) returns (ConfigGetResponse);
 
+  // Set a metadata configuration item on a project or application.
+  rpc SetMetadata(MetadataSetRequest) returns (google.protobuf.Empty);
```

Comment:
I feel like we should use `UpsertProject` and `UpsertApplication` for this and not insert a new API. This is the API that "edit project/application" uses in the UI for example and as a caller it feels weird that some fields are edited via an "update project" API and some are via a "set metadata with a oneof" API.

Reply:
Ok, that's fair.